### ISSUE TITLE
Suggestion API for many hosts, add link to the endpoint

### ISSUE THREAD

**Issue by gbin2265** [sol1/icinga-clustergraph#7]
Hello, 

I had been looking for this module for a while and found it, thx !

May I make a few suggestions for the module?

- I have more than 6000 hosts and 34 endpoints
Can you only ask the API for the attributes you need (much less information needs to be downloaded and faster reesult on the screen)
$hosts = $api->sendCommand('GET', 'objects/hosts?attrs=name&attrs=display_name&attrs=state');

- link to host for the endpoint
Once you have found the host object, maybe you can create the link to the host on the endpoint ?
for monitoring module ->  'monitoring/host/show?name=<NAME>' 
For icingadb module -> 'icingadb/host?name=<NAME>'

- I didn't understand why I always had version 1.0.5 and not 1.0.6 during installation
The version in module.info is still Version: 1.0.5

regards,
Geert

**Comment by davekempe** [comment]
@sol1-matt might have some time to have a look at this request shortly.
However I wonder if we can talk about where to draw the line here. The original purpose of this module was to show the cluster state, being only endpoints and zones. I can see a complication where having hosts on there gets very messy.

**Comment by sol1-matt** [comment]
@davekempe 
We match all endpoints to hosts by matching the hostname and endpoint name, from there we can add the hosts state to the endpoint so we have some idea if there is a problem for a given endpoint. 

There may be better ways to determine endpoint health this was just a quick and simple method.

@gbin2265 
* We need to load all hosts to gather the status info so I can see where it would get expensive, leave it with me.  
* I'll need to look into determining which modules are installed to add links, fiddle with the css so the links don't break any formatting.
* I will fix up the module.info :)

**Comment by davekempe** [comment]
Ah, so we only match the hosts with are also endpoints. I see.

**Comment by gbin2265** [comment]
Hello,

sorry for my google english :)

'.... this was just a quick and simple method.'

This was certainly a good start, but we have added this (quick and simple)

`?attrs=name&attrs=display_name&attrs=state`

to get results faster , bacause now we have to wait more than 10 seconds.
By only downloading the attributes that are needed, we now get our results much faster.

Later you can still see that you cannot query the status for each endpoint (one by one) instead of downloading all hosts.

**Comment by sol1-matt** [comment]
@gbin2265 I've made a release for the attribute filtering, and updated the module version :)

Fairly trivial change so pushing it while I work on the linking.

**Comment by sol1-matt** [comment]
@gbin2265 there is a new branch 7-links that has a implementation which has links on endpoints to the host. 

The method to handle endpoints that don't match a host is to create a link that searches for the endpoint name, a bit of a blind stab. 

There is a new config option to select if you want to use Icingaweb2 Or Icingadb Web style url's. This means systems that have both modules installed can choose which one to link to. 

Let me know if this is what you are after.

**Comment by gbin2265** [comment]
for info , I see that there are using this to see icingadb is running.

```
use Icinga\Application\Modules\Module;
use Icinga\Module\Map\ProvidedHook\Icingadb\IcingadbSupport;

        if (Module::exists('icingadb') && IcingadbSupport::useIcingaDbAsBackend()) {
           $ICINGAWEB_HOST_PATH = 'icingadb/host?name=';
        } else {
           $ICINGAWEB_HOST_PATH = 'monitoring/host/show?host=';
        }

```
!! But your solution works perfectly!

Thank you for the modification.

**Comment by sol1-matt** [comment]
released in version [v1.0.8](https://github.com/sol1/icinga-clustergraph/releases/tag/v1.0.8)

**Comment by gbin2265** [comment]
Hello,

I installed it on my production platform and everything works fine. (Speed is perfect now)

I can't think of anything that could be improved at the moment.

Thank you for the module!
